### DISCUSSION TITLE
First apply than handle in Process managers

### DISCUSSION THREAD

**Discussion by aef** [commanded/commanded#574] [Ideas]
I have a question about the behavior of process managers in Commanded. For better exposure, I asked my question [here](https://stackoverflow.com/q/50951135/322863). To me it seems that the current behavior might be a bug.

**Comment by arfl** [comment]
+1

**Comment by slashdotdash** [comment]
Process manager state should be mutated, using the `apply/2` callback function, before the domain event is processed by the `handle/2` callback.

**Comment by jdewar** [comment]
The current approach can be useful.  If `handle` blows up, I may want the process manager to be restarted with the 'previous' state; applying first would would mean I have the 'next' state.
